sdv/_utils.py

```diff
@@ -295,3 +297,50 @@ def _validate_foreign_keys_not_null(metadata, data):
             err_msg += f"Table '{table_name}', column(s) {invalid_columns}\n"
 
         raise SynthesizerInputError(err_msg)
+
+
+def check_sdv_versions_and_warn(synthesizer):
+    """Check if the current SDV and SDV Enterprise versions mismatch.
+
+    Args:
+        synthesizer (BaseSingleTableSynthesizer or BaseMultiTableSynthesizer):
+            An SDV model instance to check versions against.
+
+    Raises:
+        SDVVersionWarning:
+            If the current SDV or SDV Enterprise version does not match the version used to fit
+            the synthesizer.
+    """
+    current_public_version = getattr(version, 'public', None)
+    current_enterprise_version = getattr(version, 'enterprise', None)
+    public_missmatch = current_public_version != synthesizer._fitted_sdv_version
+    enterprise_missmatch = current_enterprise_version != synthesizer._fitted_sdv_enterprise_version
```

Comment:
I think we can't assume loaded synthesizers will have this attribute since it's being added

Reply:
This should be addressed now using `getattr` instead of directly accessing it.